src/libcprover-cpp/api.cpp

```diff
@@ -144,7 +145,9 @@ void api_sessiont::verify_model() const
   if(::process_goto_program(
        *implementation->model, *implementation->options, log))
   {
-    return;
+    // TODO: Make more robust.
+    // return;
+    INVARIANT(false, "process_goto_program failed");
```

Comment:
:raised_eyebrow: You gave this a thumbs up but haven't actioned it?

Reply:
Using `INVARIANT` here is actually a really bad thing in terms of an API. This is because if the INVARIANT fails it will either abort the whole process which is undesirable if the consumer on the API can recover or it will throw an exception which may not be supported by the ABI being used to interface with the API.

Reply:
This is no longer relevant, as marked `outdated` by the GitHub UI itself.

This change has been rolled back in b9b6cc4aa151b09f3a4407c769df4201cb4ed888

Reply:
This still needs fixing in the `produce_results` function. This comment was in the `verify_model` function but the code is duplicated, so the fix needs doing in two places.

Reply:
Good catch - my bad. I did a pattern replacement to avoid this issue, but for a strange kind of fashion this was missed.